app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java

```diff
@@ -44,6 +53,20 @@ public CompletableFuture updateFavor(Favor favor) {
     return getRepository().updateFavor(favor);
   }
 
+  // Upload/download pictures
+  @Override
+  public void uploadOrUpdatePicture(Favor favor, Bitmap picture) {
+    CompletableFuture<String> pictureUrl = getPictureUtility().uploadPicture(picture);
+    pictureUrl.thenAccept(url -> FavorUtil.getSingleInstance().updateFavorPhoto(favor, url));
+  }
+
+  @Override
+  public CompletableFuture<Bitmap> downloadPicture(Favor favor) throws RuntimeException {
+    String url = favor.getPictureUrl();
+    if (url == null) throw new RuntimeException("Invalid picture url in Favor");

```

Comment:
instead of throwing you should complete a future exceptionally

Reply:
Good idea!